x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/Version.java

```diff
@@ -86,26 +87,34 @@ public static Version fromString(String version) {
         // This is similar to how Elasticsearch's Build class digs up its build information.
         // Since version info is not critical, the parsing is lenient
         URL url = Version.class.getProtectionDomain().getCodeSource().getLocation();
-        String urlStr = url.toString();
+        CURRENT = extractVersion(url);
+    }
 
+    static Version extractVersion(URL url) {
+        String urlStr = url.toString();
         byte maj = 0, min = 0, rev = 0;
         String ver = "Unknown";
         String hash = ver;
-
-        if (urlStr.endsWith(".jar")) {
-            try (JarInputStream jar = new JarInputStream(url.openStream())) {
-                Manifest manifest = jar.getManifest();
-                hash = manifest.getMainAttributes().getValue("Change");
-                ver = manifest.getMainAttributes().getValue("X-Compile-Elasticsearch-Version");
-                byte[] vers = from(ver);
-                maj = vers[0];
-                min = vers[1];
-                rev = vers[2];
+        
+        if (urlStr.endsWith(".jar") || urlStr.endsWith(".jar!/")) {
+            try {
+                URLConnection conn = url.openConnection();
+                conn.setUseCaches(false);
+                
+                try (JarInputStream jar = new JarInputStream(conn.getInputStream())) {
+                    Manifest manifest = jar.getManifest();
+                    hash = manifest.getMainAttributes().getValue("Change");
+                    ver = manifest.getMainAttributes().getValue("X-Compile-Elasticsearch-Version");
+                    byte[] vers = from(ver);
```

Comment:
I know this is unmodified code, however, it's noteworthy that [server/src/main/java/org/elasticsearch/Version.java](https://github.com/elastic/elasticsearch/blob/12b24bfa57ae3b282427298cffc735ab9fe5868f/server/src/main/java/org/elasticsearch/Version.java#L98) parses ints, not bytes. Not sure if that's because it later does some int maths with the components, or because it reserves the possibility to use some tagging values at some point (like `7.2.222` for a special release). Just mentioning it, should we ever want to align this.